using-vertx/src/main/java/org/acme/vertx/FruitResource.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.acme.vertx;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+import javax.inject.Inject;
+import javax.ws.rs.*;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.ResponseBuilder;
+import javax.ws.rs.core.Response.Status;
+
+import org.jboss.resteasy.annotations.jaxrs.PathParam;
+
+import io.reactiverse.axle.pgclient.*;
+
+@Path("fruits")
+@Produces("application/json")
+@Consumes("application/json")
+public class FruitResource {
+
+    @Inject
+    PgPool client;
+
+    @GET
+    public CompletionStage<Response> get() {
+        CompletionStage<PgRowSet> rowSet = client.query("SELECT id, name FROM fruits ORDER BY name ASC");
+        CompletionStage<List<Fruit>> fruits = rowSet.thenApply(pgRowSet -> {
+            List<Fruit> list = new ArrayList<>(pgRowSet.size());
+            PgIterator pgIterator = pgRowSet.iterator();
+            while (pgIterator.hasNext()) {
+                list.add(toFruit(pgIterator.next()));
+            }
+            return list;
+        });
+        CompletionStage<Response> response = fruits
+                .thenApply(list -> list.isEmpty() ? Response.noContent() : Response.ok(list))
+                .thenApply(ResponseBuilder::build);
+        return response;
+    }
+
+    @GET
+    @Path("{id}")
+    public CompletionStage<Response> getSingle(@PathParam("id") Long id) {
```

Comment:
you're using the right annotation: if you configure your maven compiler to include method parameter names you can just use `@PathParam` with no parameter.

Reply:
But then we need to tell the user to do this in the guide, don't we? For people following the guide and just looking at the quickstart it will be fine though.

Reply:
We should tell them, yes.